src/Mailgun/Api/MailingList.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+/*
+ * Copyright (C) 2013 Mailgun
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+namespace Mailgun\Api;
+
+use Mailgun\Assert;
+use Mailgun\Model\MailingList\PagesResponse;
+use Mailgun\Model\MailingList\CreateResponse;
+use Mailgun\Model\MailingList\ShowResponse;
+use Mailgun\Model\MailingList\UpdateResponse;
+use Mailgun\Model\MailingList\DeleteResponse;
+use Mailgun\Model\MailingList\MembersResponse;
+use Mailgun\Model\MailingList\CreateMemberResponse;
+use Mailgun\Model\MailingList\ShowMemberResponse;
+use Mailgun\Model\MailingList\UpdateMemberResponse;
+use Mailgun\Model\MailingList\DeleteMemberResponse;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * @author Michael Münch <helmchen@sounds-like.me>
+ */
+class MailingList extends HttpApi
+{
+    /**
+     * Returns a paginated list of mailing lists on the domain.
+     *
+     * @param int $limit Maximum number of records to return (optional: 100 by default)
+     *
+     * @return PagesResponse|array|ResponseInterface
```

Comment:
What about if you use the ArrayHydrator? :)

Reply:
I don't think it might be the case based on the current code, our Hydrator returns a `ResponseInterface` that it's implemented by our models so in the end we will get an instance of the object that we're passing as an argument of the Hydrator.

Reply:
```
final class ArrayHydrator implements Hydrator
{
    /**
     * @param ResponseInterface $response
     * @param string            $class
     *
     * @return array // <==
     */
```

I can adjust the @return statements that is not a problem, I was just not sure because I had seen that in the other classes as well.

Reply:
🤔 Nice point, I'll take a look to confirm

Reply:
I think it won't be a problem because, for this case, we use the Model Hydrator, meaning that the provided class is the one that sets the output:

https://github.com/mailgun/mailgun-php/blob/master/src/Mailgun/Hydrator/ModelHydrator.php#L42-L46

Reply:
the default Hydrator is the ModelHydrator yes .. but we can override the Hydrator during the construction of the Mailgun Object and if so, the return type will vary between a Response implementing ResponseInterface and a plain array in case of the ArrayHydrator. Or did i miss something here? 

I do not want to be picky right now, just to understand the whole thing :)

Reply:
Hey @Etmolf 

I am not sure to follow you, because we're expecting to create the Mailgun object through the `create` static method (as described on [Usage](https://github.com/mailgun/mailgun-php/blob/master/README.md#usage)'s section on README file), that is expecting just 1 argument: the API Key

https://github.com/mailgun/mailgun-php/blob/master/src/Mailgun/Mailgun.php#L105-L115

Original Mailgun's constructor (that was allowing you to customise the Hydrator)  is no longer recommended to instantiate the Mailgun's client, and 2 of the expected arguments have been already deprecated

https://github.com/mailgun/mailgun-php/blob/master/src/Mailgun/Mailgun.php#L64-L86

Based on this, I believe that we can work thinking about the `ModelHydrator` with no other conditionals in mind, that - in fact - is set automatically on the existing public constructor

https://github.com/mailgun/mailgun-php/blob/master/src/Mailgun/Mailgun.php#L85

Hope this makes sense to you

Reply:
Alright :) If you put it that way, that makes sense.

I change the annotations accordingly